flask_admin/templates/bootstrap3/admin/lib.html

```diff
@@ -169,7 +169,7 @@
   <link href="{{ admin_static.url(filename='vendor/select2/select2.css') }}" rel="stylesheet">
   <link href="{{ admin_static.url(filename='vendor/select2/select2-bootstrap3.css') }}" rel="stylesheet">
   <link href="{{ admin_static.url(filename='vendor/bootstrap-daterangepicker/daterangepicker-bs3.css') }}" rel="stylesheet">
-  {% if config.MAPBOX_MAP_ID %}
+  {% if config.MAPBOX_MAP_ID and config.MAPBOX_ACCESS_TOKEN %}
```

Comment:
This will prevent inclusion of the leaflet assets even for the v3 API, since MAPBOX_ACCESS_TOKEN isn't expected in the config in that case.